src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -107,20 +107,70 @@ public async Task<ExecutionResult> ExecuteAsync(ExecutionOptions options)
                         _complexityAnalyzer.Validate(document, options.ComplexityConfiguration);
                 }
 
-                context = BuildExecutionContext(
-                    options.Schema,
-                    options.Root,
-                    document,
-                    operation,
-                    options.Inputs,
-                    options.UserContext,
-                    options.CancellationToken,
-                    metrics,
-                    options.Listeners,
-                    options.ThrowOnUnhandledException,
-                    options.UnhandledExceptionDelegate,
-                    options.MaxParallelExecutionCount,
-                    options.RequestServices);
+                try
+                {
+                    context = BuildExecutionContext(
+                        options.Schema,
+                        options.Root,
+                        document,
+                        operation,
+                        options.Inputs ?? Inputs.Empty,
+                        options.UserContext,
+                        options.CancellationToken,
+                        metrics,
+                        options.Listeners,
+                        options.ThrowOnUnhandledException,
+                        options.UnhandledExceptionDelegate,
+                        options.MaxParallelExecutionCount,
+                        options.RequestServices);
+                }
+                catch
```

Comment:
Catch should work with more specific exception.

Reply:
Should be fine.

```suggestion
                catch (InvalidVariableError)
```